examples/combobox/js/combobox-autocomplete.js

```diff
@@ -0,0 +1,558 @@
+  this.comboboxNode.parentNode.classList.add('focus'); // set the focus class to the parent for easier styling
+  this.comboboxHasVisualFocus = true;
+  this.listboxHasVisualFocus = false;
+  this.setActiveDescendant(false);
+};
+
+ComboboxAutocomplete.prototype.setVisualFocusListbox = function () {
+  this.comboboxNode.parentNode.classList.remove('focus');
+  this.comboboxHasVisualFocus = false;
+  this.listboxHasVisualFocus = true;
+  this.listboxNode.classList.add('focus');
+  this.setActiveDescendant(this.option);
+};
+
+ComboboxAutocomplete.prototype.removeVisualFocusAll = function () {
+  this.comboboxNode.parentNode.classList.remove('focus');
+  this.comboboxHasVisualFocus = false;
+  this.listboxHasVisualFocus = false;
+  this.listboxNode.classList.remove('focus');
+  this.option = false;
+  this.setActiveDescendant(false);
+};
+
+// ComboboxAutocomplete Events
+
+ComboboxAutocomplete.prototype.filterOptions = function () {
+
+  // do not filter any options if autocomplete is none
+  if (this.isNone) {
+    this.filter = '';
+  }
+
+  var currentOption = this.option;
+  var filter = this.filter.toLowerCase();
+
+  this.filteredOptions    = [];
+  this.listboxNode.innerHTML = '';
+
+  for (var i = 0; i < this.allOptions.length; i++) {
+    var option = this.allOptions[i];
+    if (filter.length === 0 || option.textComparison.indexOf(filter) === 0) {
+      this.filteredOptions.push(option);
+      var textContent = option.textContent.trim();
+      this.listboxNode.appendChild(option);
+    }
+  }
+
+  // Use populated options array to initialize firstOption and lastOption.
+  var numItems = this.filteredOptions.length;
+  if (numItems > 0) {
+    this.firstOption = this.filteredOptions[0];
+    this.lastOption  = this.filteredOptions[numItems - 1];
+
+    if (currentOption && this.filteredOptions.indexOf(currentOption) >= 0) {
+      option = currentOption;
+    }
+    else {
+      option = this.firstOption;
+    }
+  }
+  else {
+    this.firstOption = false;
+    option = false;
+    this.lastOption  = false;
+  }
+
+  return option;
+};
+
+ComboboxAutocomplete.prototype.setCurrentOptionStyle = function (option) {
+
+  for (var i = 0; i < this.filteredOptions.length; i++) {
+    var opt = this.filteredOptions[i];
+    if (opt === option) {
+      opt.setAttribute('aria-selected', 'true');
+      while ((this.listboxNode.scrollTop + this.listboxNode.offsetHeight) < (opt.offsetTop + opt.offsetHeight)) {
+        this.listboxNode.scrollTop = this.listboxNode.scrollTop + opt.offsetHeight;
+      }
+      while (this.listboxNode.scrollTop > (opt.offsetTop + 2)) {
+        this.listboxNode.scrollTop = this.listboxNode.scrollTop - opt.offsetHeight;
+      }
+    }
+    else {
+      opt.removeAttribute('aria-selected');
+    }
+  }
+};
+
+ComboboxAutocomplete.prototype.getPreviousOption = function (currentOption) {
+  var index;
+
+  if (currentOption !== this.firstOption) {
+    index = this.filteredOptions.indexOf(currentOption);
```

Comment:
`index` can be declared here, since it's only used within the `if` block

Reply:
Fixed in 5be02daa88b9c402a9a31bdca5ef95c3b03956ab